source/includes/fundamentals/code-snippets/builders/AggBuilders.java

```diff
@@ -0,0 +1,287 @@
+        // end graphLookupBasic
+    }
+
+    private void mergeOptionsStage() {
+        // begin mergeOptions
+        merge(new MongoNamespace("reporting", "customers"),
+                new MergeOptions().uniqueIdentifier(asList("date", "customerId"))
+                        .whenMatched(MergeOptions.WhenMatched.REPLACE)
+                        .whenNotMatched(MergeOptions.WhenNotMatched.INSERT));
+        // end mergeOptions
+    }
+
+    private void mergeStage() {
+        // begin mergeStage
+        merge("authors");
+        // end mergeStage
+    }
+
+    private void outStage() {
+        // begin out
+        out("authors");
+        // end out
+    }
+
+    private void unwindIndexStage() {
+        // begin unwindIndex
+        unwind("$sizes", new UnwindOptions().includeArrayIndex("position"));
+        // end unwindIndex
+    }
+
+    private void unwindPreserveStage() {
+        // begin unwindPreserve
+        unwind("$sizes", new UnwindOptions().preserveNullAndEmptyArrays(true));
+        // end unwindPreserve
+    }
+
+    private void unwindStage() {
+        // begin unwindStage
+        unwind("$sizes");
+        // end unwindStage
+    }
+
+    private void groupStage() {
+        // begin group
+        group("$customerId", Accumulators.sum("totalQuantity", "$quantity"),
+                Accumulators.avg("averageQuantity", "$quantity"));
+        // end group
+    }
+
+    private void advancedLookupStage() {
+        database = mongoClient.getDatabase("agg-builders");
+        collection = database.getCollection("warehouses");
+        collection.drop();
+        collection.insertMany(asList(
+                new Document("_id", 1).append("stock_item", "almonds").append("warehouse", "A").append("instock", 120),
+                new Document("_id", 2).append("stock_item", "pecans").append("warehouse", "A").append("instock", 80),
+                new Document("_id", 3).append("stock_item", "almonds").append("warehouse", "B").append("instock", 60),
+                new Document("_id", 4).append("stock_item", "cookies").append("warehouse", "B").append("instock", 40),
+                new Document("_id", 5).append("stock_item", "cookies").append("warehouse", "A").append("instock", 80)));
+        collection = database.getCollection("orders");
+        collection.drop();
+        collection.insertMany(
+                asList(new Document("_id", 1).append("item", "almonds").append("price", 12).append("ordered", 2),
+                        new Document("_id", 2).append("item", "pecans").append("price", 20).append("ordered", 1),
+                        new Document("_id", 3).append("item", "cookies").append("price", 10).append("ordered", 60)));
+        // begin advanced lookup
+        List<Variable<String>> variables = asList(new Variable<>("order_item", "$item"),
+                new Variable<>("order_qty", "$ordered"));
+
+        List<Bson> pipeline = asList(
```

Comment:
It's a bit jarring as a reader to find the first example of a list of pipeline stages, as well as its use in the CRUD API to be in an example that appears right in the middle of all the other examples.  I know there is a link to the aggregation guide at the top, which does show a complete example, but still, it's awkward.

I won't insist on a change as it's not technically incorrect, but consider the following suggestions.

* Show one example at the very top when a stage is is added to a List
* Remove the use of collection.aggregate in this example.